rotkehlchen/chain/ethereum/uniswap/typing.py

```diff
@@ -74,6 +85,273 @@ class ProtocolBalance(NamedTuple):
     unknown_assets: Set[UnknownEthereumToken]
 
 
-# Get History
+# Get trades history
 
 AddressTrades = Dict[ChecksumEthAddress, List[AMMTrade]]
+ProtocolHistory = Dict[str, Union[AddressTrades]]
+
+
+# Get events history
+
+
+class EventType(Enum):
+    """Supported Locations"""
+    MINT = 1
+    BURN = 2
+
+    def __str__(self) -> str:
+        if self == EventType.MINT:
+            return 'mint'
+        elif self == EventType.BURN:
+            return 'burn'
+
+        raise RuntimeError(f'Corrupt value {self} for EventType -- Should never happen')
+
+    @classmethod
+    def deserialize_from_db(cls, symbol: str) -> 'EventType':
```

Comment:
But even if I'm storing just a string, imho it is worth it to have the enum structure. It fits as a specific constant set of options and I do care about the value. In fact I'm constantly using it for processing the response data and for calculating the profit/loss. How would you do it simpler and nicer, dealing with deserialization errors and not having all the functionality scattered around? 2 types, but 30 lines is not that bloated, no?

Reply:
Oh yes have the enum data class.

 But not the serialization/derialization to the DB as 'A' and 'B' since that's an overkill and as per the other comment and would also need a special enum table in the DB.

The comment I am making here is specifically on the "deserialize_from_db()" function.

Reply:
![2020-11-18_13-45](https://user-images.githubusercontent.com/1658405/99532322-62915880-29a4-11eb-92a5-ce80130e18fa.png)